## Задача 1.
_Даны две независимые выборки. Не соблюдается условие нормальности_

$x1: 380, 420, 290$<br>
$y1: 140, 360, 200, 900$

_Сделайте вывод по результатам, полученным с помощью функции_

Используем критерий Манна – Уитни.

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$

In [1]:
from scipy import stats
import numpy as np

In [2]:
x = np.array([380, 420, 290])
y = np.array([140, 360, 200, 900])

In [3]:
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

$pvalue$ больше любых стандартных $\alpha$, следовательно нулевая гипотеза об отсутствии статистически значимых различий принимается.

## Задача 2.
_Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут._
_Есть ли статистически значимые различия?_


$1$е измерение до приема препарата: $150, 160, 165, 145, 155$<br>
$2$е измерение через $10$ минут: $140, 155, 150, 130, 135$<br>
$3$е измерение через $30$ минут: $130, 130, 120, 130, 125$<br>


Используем критерий Фридмана.

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [4]:
a = np.array ([150, 160, 165, 145, 155])
b = np.array ([140, 155, 150, 130, 135])
c = np.array ([130, 130, 120, 130, 125])

In [5]:
stats.friedmanchisquare(a, b, c)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

$pvalue < \alpha\$

$0.0083 < 0.05$

т.о. нулевая гипотеза об отсутствии статистически значимых различий отвергается. Статистически значимые различия есть.

## Задача 3.
_Сравните $1$ и $2$-е измерения, предполагая, что $3$го измерения через $30$ минут не было._

Используем критерий Уилкоксона.

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [6]:
stats.wilcoxon(a, b)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

$pvalue > \alpha\$

$0.0625 > 0.05$

т.о. нулевая гипотеза об отсутствии статистически значимых различий принимается. Статистически значимых различий нет.

## Задача 4.

_Даны $3$ группы учеников плавания._

В $1$ группе время на дистанцию $50$ м составляют: $56, 60, 62, 55, 71, 67, 59, 58, 64, 67$<br>
Вторая группа: $57, 58, 69, 48, 72, 70, 68, 71, 50, 53$<br>
Третья группа: $57, 67, 49, 48, 47, 55, 66, 51, 54$<br>

Используем критерий Крускала – Уоллиса.

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [7]:
g1 =([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
g2 = ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
g3 = ([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(g1, g2, g3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

$pvalue > \alpha\$

$0.065 > 0.05$

т.о. нулевая гипотеза об отсутствии статистически значимых различий принимается.Статистически значимых различий нет.

## Задача 5.
_Заявляется, что партия изготавливается со средним арифметическим $2,5$ см.
Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения.Объем выборки $10$, уровень статистической значимости $5\%$_

$2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34$

$H_0: \mu_x = 2.5$<br>
$H_1: \mu_x \neq 2.5$

 $\alpha = 0.05$

$\sigma =\frac{1}{n-1}\sum(X_i-\overline{X})^2$

$t=\frac{\overline{X}-\mu}{\frac{\sigma_н}{\sqrt{n}}}$

In [8]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mean_X = X.mean()
std_X = X.std(ddof=1)
t_fact = (mean_X - 2.5) / std_X * np.sqrt(10)
t_fact

0.563061366180296

По таблице критических точек распределения Стьюдента для двухсторонней области найдем критическую точку по уровню значимости α = 0,05 и числу степеней свободы k = 9.

In [9]:
stats.t.ppf(1 - 0.05 / 2, 9)

2.2621571627409915

$t_{crit}$ = 2.26

$t_{fact}$ = 0.563

$|t_{fact}|<t_{crit}$

т.о. нулевая гипотеза принимается. Партия изготавливается со средним арифметическим 2,5 см.